src/atfile.c

```diff
@@ -138,11 +145,13 @@ CAMLprim value caml_extunix_linkat(value v_olddirfd, value v_oldname, value v_ne
 CAMLprim value caml_extunix_fchownat(value v_dirfd, value v_name, value v_owner, value v_group, value v_flags)
 {
   CAMLparam5(v_dirfd, v_name, v_owner, v_group, v_flags);
+  int dirfd = Int_val(v_dirfd), owner = Int_val(v_owner), group = Int_val(v_group);
+  const char *name = String_val(v_name);
   int ret = 0;
   int flags = caml_convert_flag_list(v_flags, at_flags_table);
   flags &= (AT_SYMLINK_NOFOLLOW /* | AT_EMPTY_PATH */);  /* only allowed flag here */
   caml_enter_blocking_section();
-  ret = fchownat(Int_val(v_dirfd), String_val(v_name), Int_val(v_owner), Int_val(v_group), flags);
```

Comment:
Int_val is safe, I would be ok to keep to not pollute the scope, but work already done so I guess it is ok

Reply:
My understanding is that it is not safe to use `Int_val` here inside of a blocking section as the values have been registered as roots. See https://github.com/ocaml/ocaml/pull/13188#issuecomment-2124225485 and https://github.com/ocaml/ocaml/pull/13188#issuecomment-2124287000.

Reply:
yeah, this is recent development, i hope upstream can find better solution, so annoying